File: Java/SpectableJavaExamplesTest/src/test/java/spectable/common/PricingString.java
package spectable.common;

import java.util.Objects;
import production.*;

public class PricingString {
    private static final String DNCString = "?DNC?";

    public String totalPrice;

    public PricingString(String totalPrice) {
        this.totalPrice = totalPrice;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof PricingString)) return false;
        PricingString that = (PricingString) o;
        return (DNCString.equals(totalPrice) || DNCString.equals(that.totalPrice) || Objects.equals(totalPrice, that.totalPrice));
    }

    @Override
    public int hashCode() {
        return Objects.hash(totalPrice);
    }

    @Override
    public String toString() {
        return "TotalPrice=" + totalPrice;
    }
}

File: CSharp/SpecStudioExamplesTestsCSharp/common/PricingString.cs
namespace SpecStudioExamplesTestsCSharp.common
{
using production;

    public class PricingString
    {
        public string totalPrice;

        public PricingString(string totalPrice)
        {
            this.totalPrice = totalPrice;
        }

        public PricingTyped ToPricingTyped()
        {
            return new PricingTyped(
                new Dollar(this.totalPrice)
            );
        }

        public override string ToString()
        {
            return $"TotalPrice={totalPrice}";
        }
    }
}
